src/main/org/deidentifier/arx/algorithm/DataDependentEDDPAlgorithm.java

```diff
@@ -0,0 +1,183 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.deidentifier.arx.algorithm;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.math3.fraction.BigFraction;
+import org.deidentifier.arx.dp.ExponentialMechanismReliable;
+import org.deidentifier.arx.framework.check.TransformationChecker;
+import org.deidentifier.arx.framework.check.TransformationChecker.ScoreType;
+import org.deidentifier.arx.framework.check.history.History.StorageStrategy;
+import org.deidentifier.arx.framework.lattice.SolutionSpace;
+import org.deidentifier.arx.framework.lattice.Transformation;
+import org.deidentifier.arx.metric.v2.ILScore;
+import org.deidentifier.arx.reliability.IntervalArithmeticDouble;
+import org.deidentifier.arx.reliability.IntervalArithmeticException;
+
+import cern.colt.list.LongArrayList;
+import de.linearbits.jhpl.PredictiveProperty;
+
+/**
+ * This class implements the search algorithm used with data-dependent differential privacy
+ * 
+ * @author Raffael Bild
+ */
+public class DataDependentEDDPAlgorithm extends AbstractAlgorithm {
+    
+    /** Property */
+    private final PredictiveProperty                 propertyChecked;
+
+    /** Number of steps to be performed */
+    private final int                                steps;
+
+    /** The expopnential mechanism */
+    private final ExponentialMechanismReliable<Long> exponentialMechanism;
+
+    /**
+     * Creates a new instance
+     * @param solutionSpace
+     * @param checker
+     * @param deterministic
+     * @param steps
+     * @param epsilonSearch
+     * @return
+     */
+    public static AbstractAlgorithm create(SolutionSpace solutionSpace, TransformationChecker checker,
+                                           boolean deterministic, int steps, double epsilonSearch) {
+        return new DataDependentEDDPAlgorithm(solutionSpace, checker, deterministic, steps, epsilonSearch);
+    }
+
+    /**
+     * Constructor
+     * @param space
+     * @param checker
+     * @param deterministic
+     * @param steps
+     * @param epsilonSearch
+     */
+    private DataDependentEDDPAlgorithm(SolutionSpace space, TransformationChecker checker,
+                                       boolean deterministic, int steps, double epsilonSearch) {
+        super(space, checker);
+        this.checker.getHistory().setStorageStrategy(StorageStrategy.ALL);
+        this.propertyChecked = space.getPropertyChecked();
+        this.solutionSpace.setAnonymityPropertyPredictable(false);
+        this.steps = steps;
+
+        double epsilonPerStep;
+        IntervalArithmeticDouble arithmetic = new IntervalArithmeticDouble();
+        try {
+            epsilonPerStep = arithmetic.div(arithmetic.createInterval(epsilonSearch), arithmetic.createInterval(steps)).lower;
+            exponentialMechanism = new ExponentialMechanismReliable<Long>(epsilonPerStep, deterministic);

```

Comment:
We don't include a non-reliable variant of the Expontential Mechanism, right? This means that the class ExponentialMechanismReliable can be renamed to ExponentialMechanism.